File: 0048-rotate-image/0048-rotate-image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var trans = function(matrix)  {
    
    for(let i = 0; i<matrix.length; i++){
        for(let j = i+1; j < matrix[i].length; j++){
            [matrix[i][j],matrix[j][i]] = [matrix[j][i],matrix[i][j]];
        }
    }
}
var rotate = function(matrix) {
    
    trans(matrix);
    for(let i = 0; i<matrix.length; i++){
        matrix[i].reverse();
    }
};

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
private:
    void tranv(vector<vector<int>>&m){
        for(int i = 0 ; i<m.size() ; i++){
            for(int j = i+1 ; j < m[0].size() ; j++){
                swap(m[i][j],m[j][i]);
            }
        }
        
    }
public:
    void rotate(vector<vector<int>>& matrix) {
        
        tranv(matrix);
        for(int i = 0 ; i < matrix.size() ; i++)
            reverse(matrix[i].begin(),matrix[i].end());
    }
};